### DISCUSSION TITLE
Flatpak module yaml file for PyMuPDF

### DISCUSSION THREAD

**Discussion by cbm755** [pymupdf/PyMuPDF#1861] [Installation Topics]
If anyone needs PyMuPDF as a dependency for a [Flatpak](https://flatpak.org/), feel free to use this module:

https://github.com/flathub/org.plomgrading.PlomClient/blob/master/python3-pymupdf.yaml

My intention will be to update this after new PyMuPDF releases, although perhaps the above link will track whatever version Plom is pinned.  If *you* update it first, please post below.

- - - -

### background

I needed this for the linked project, and found the [usual tooling](https://github.com/flatpak/flatpak-builder-tools) namely `flatpak-pip-generator` was trying to build PyMuPDF from source, which is a pain b/c then we need MuPDF from source etc etc.

Without this, my builds would fail like:
```
  PyMuPDF/setup.py:     PIP_EXISTS_ACTION: i
  PyMuPDF/setup.py:     PIP_BUILD_TRACKER: /tmp/pip-build-tracker-sbu00e44
  PyMuPDF/setup.py:     PLAT: linux-aarch64
  running install
  /usr/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
    warnings.warn(
  running build
  running build_py
  running build_ext
  building 'fitz._fitz' extension
  swigging fitz/fitz.i to fitz/fitz_wrap.c
  swig -python -o fitz/fitz_wrap.c fitz/fitz.i
  error: command 'swig' failed: No such file or directory
  error: subprocess-exited-with-error
  ```

Instead I found https://github.com/flatpak/flatpak-builder-tools/issues/296#issuecomment-1173037783
which showed me how to specify all the relevant wheels manually.  thanks @real-yfprojects.

This involved a bunch of copy-paste so I thought I'd share it here for search engines to index for the benefit of others.

License is FSF All-Permissive (FSFAP).

**Comment by cbm755** [comment]
I have updated this for 1.20.2, same link as above.

**Comment by cbm755** [comment]
Updated again for 1.21.1.

**Reply by real-yfprojects** [reply]
Have you seen [req2flatpak.py](https://github.com/johannesjh/req2flatpak)?

**Reply by cbm755** [reply]
very cool, will look into this next time I'm updating my Flatpak, or of course patches always welcome

https://github.com/flathub/org.plomgrading.PlomClient/issues/79

**Reply by cbm755** [reply]
Thanks @real-yfprojects, I have started using that so I will not maintain my manually-generated .yaml any more. 

- - - -
 
If anyone was relying on that, I suggest:
```
pip install req2flatpak
req2flatpak --requirements pymupdf==1.21.1 --target-platforms 39-x86_64 39-aarch64 --outfile python3-pymupdf.json
```
The `39` refers to Python 3.9 which in my case is what my Flatpak base image is using.  For example, for Python 3.10: `... --target-platforms 310-x86_64 310-aarch64`

- - - - 

FYI: I think I could go further and use `req2flatpak` directly on my `requirements.txt`: maybe I'll do so on a future release.

**Reply by real-yfprojects** [reply]
Great! If you encounter any problem like missing yaml support, feel free to tell me or open an issue.
